pkg/operator/controller/extension/runtime/runtime.go

```diff
@@ -0,0 +1,136 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package runtime
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1alpha1 "github.com/gardener/gardener/pkg/apis/operator/v1alpha1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/utils"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+	"github.com/gardener/gardener/pkg/utils/oci"
+)
+
+// Interface contains functions for the extension deployment in the garden runtime cluster.
+type Interface interface {
+	// Reconcile creates or updates the extension deployment in the garden runtime cluster.
+	Reconcile(context.Context, logr.Logger, *operatorv1alpha1.Extension) error
+	// Delete deletes the extension deployment in the garden runtime cluster.
+	Delete(context.Context, logr.Logger, *operatorv1alpha1.Extension) error
+}
+
+type deployment struct {
+	runtimeClientSet kubernetes.Interface
+	recorder         record.EventRecorder
+
+	gardenNamespace string
+	helmRegistry    oci.Interface
+}
+
+// Reconcile creates or updates the extension deployment in the garden runtime cluster.
+// If the extension doesn't define an extension deployment for the runtime cluster, the deployment is deleted.
+func (d *deployment) Reconcile(ctx context.Context, log logr.Logger, extension *operatorv1alpha1.Extension) error {
+	if extensionDeploymentSpecified(extension) {
+		log.Info("Deploying extension resources in garden runtime cluster")
```

Comment:
I think, [this](https://github.com/gardener/gardener/blob/aeb5571a35c9689ec48c5b28903cabcc018e582f/pkg/operator/controller/extension/reconciler.go#L71) might be the magic place where it happens 😄

Reply:
I do not get it. `logf.FromContext(ctx)` does not call `log.WithValues(...)` with the extension name.
If the extension name is showing up in the logs I am fine with it, but it would nevertheless be interesting how it works. As far as I can tell, there should be no extension name in the logs. Do you have some logs of an extension reconciliation handy?

Reply:
Not exactly for this log line, but for others of the same controller which uses the same logger ([ref](https://github.com/gardener/gardener/blob/a568bbc295de3e53cc59ebbd1d4d4c6f4175c5a4/pkg/operator/controller/extension/admission/admission.go#L51), [ref](https://github.com/gardener/gardener/blob/a568bbc295de3e53cc59ebbd1d4d4c6f4175c5a4/pkg/operator/controller/extension/admission/admission.go#L64))

```
{"level":"info","ts":"2024-09-20T11:36:06.622Z","msg":"Deploying admission virtual garden resources","controller":"extension","controllerGroup":"operator.gardener.cloud","controllerKind":"Extension","Extension":{"name":"provider-local"},"namespace":"","name":"provider-local","reconcileID":"a6bab46f-1f40-48d8-bcf6-2dd8e0d6a61b"}
{"level":"info","ts":"2024-09-20T11:36:06.623Z","msg":"Deploying admission runtime resources","controller":"extension","controllerGroup":"operator.gardener.cloud","controllerKind":"Extension","Extension":{"name":"provider-local"},"namespace":"","name":"provider-local","reconcileID":"a6bab46f-1f40-48d8-bcf6-2dd8e0d6a61b"}
```